### DISCUSSION TITLE
Multicores

### DISCUSSION THREAD

**Discussion by odrya12** [serverpod/serverpod#1306] [Q&A] [answered]
Hi,

I could not found any explanation in the docs....

Does Serverpod use multi-core to handle multi-requests in the same time or I need to manage it with some load-balancer ?

If it does:
How It decide how many cores to use?
Can I configure it?

Thanks, Omer Ashkenazi.

**Comment by odrya12** [comment]
After searching, it looks like serverpod doesn't set 'shared' to true.

```dart
var httpServer = await HttpServer.bindSecure(
  InternetAddress.anyIPv6,
  port,
  securityContext!,
);
```

It means you indeed need to manage multiple requests by yourself. (I may be wrong)

Hope the dev team will confirm with me on that. I think any developer should know about it.

**Reply by Isakdl** [reply, marked as answer]
No, you do not need to manage this!

In Dart you have the event loop, and no async call is blocking like in other programming languages such as Java. This means that even a single thread can handle multiple requests just by virtue of running in Dart. Take a look here for more detailed information on how this works: https://dart.dev/language/concurrency.
